### PR TITLE
Support v3 and v3.1

### PR DESCRIPTION
**Pull request by latanasov**
Added support for v3 and v3.1 send API + bug fixes.

### CODE REVIEW

**Comment by Nightbr** [comment]
Hey!
Amazing work!! It sounds really great!

I have some review:

* I would create a folder `MessageFormat` which contains:
    - `MessageFormatStrategyInterface.php` (and remove the interface from MailjetTransport)
    - `MessagePayloadV3.php` (don't forget the uppercase at first caracter because it's a class name)
    - `MessagePayloadV31.php`
    - You can also put the `Utils.php` in this folder or create a `BaseMessagePayload.php` which implements `MessageFormatStrategyInterface` and all MessagePayloadXXX will extends from BaseMessagePayload.

In order to respect the [SRP](https://en.wikipedia.org/wiki/Single_responsibility_principle).

**Comment by latanasov** [comment]
Hi @Nightbr , thank you for your suggestions- they are good and will be implemented :)

**Comment by Nightbr** [comment]
Really great implementation BTW, I really love the way it automatically format the payload for API V3 or V3.1 👍 

I just made some code review (doc, ...) just few minor things to fix.

Add to that, I have created some test.json in the folder `Tests/Resources`, I think we can retrieve them and put them into subfolder such as:

* `Tests\Resources\V3`: https://github.com/mailjet/MailjetSwiftMailer/tree/master/Tests/Resources
* `Tests\Resources\V31`: https://github.com/mailjet/MailjetSwiftMailer/tree/sendAPI3.1/Tests/Resources

It could be used for more unit tests.

**Comment by latanasov** [comment]
@Nightbr your suggestions are very appreciated, already working on them :)

File: README.md [hunk 1]
```diff
@@ -6,11 +6,17 @@
 [![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/mailjet/MailjetSwiftMailer/blob/master/LICENSE.md)
 
 A SwiftMailer transport implementation for Mailjet
-
-*Compatible Mailjet send API V3*
+([NEW] we now support send API v3.1 )
```
Comment (arnaudbreton) [inline README.md:9]: Add a link pointing to https://dev.mailjet.com/guides/#send-api-v3-1-beta

File: README.md [hunk 2]
```diff
@@ -23,22 +29,26 @@ composer require mailjet/mailjet-swiftmailer
 
 ```php
 $transport = new MailjetTransport($dispatchEvent, $apiKey, $apiSecret);
-$transport->setClientOptions(['url' => "www.mailjet.com", 'version' => 'v3', 'call' => true]); // optional
+$transport->setClientOptions(['url' => "api.mailjet.com", 'version' => 'v3.1', 'call' => true]); // optional
```
Comment (arnaudbreton) [inline README.md]: Why is it optional? It depends on which Send API version you're willing to choose, right?

File: Tests/SwiftMailer/MailjetTransportv31Test.php [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+                <p>My first paragraph.</p>
+
+                </body>
+                </html>
+                ", "text/html");
+        $mailjetMessage = $transport->messageFormat->getMailjetMessage($message)['Messages'][0];
+        $result = $transport->send($message);
+
+        $this->assertEquals("<!DOCTYPE html>
```
Comment (arnaudbreton) [inline Tests/SwiftMailer/MailjetTransportv31Test.php:83]: Better store it in a constant so if you update the `setBody`, you do not have to update the `assertEquals`?

File: Tests/SwiftMailer/MailjetTransportv31Test.php [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+        $result = $transport->send($message);
+
+        $this->assertEquals('azertyuiop', $mailjetMessage['TemplateID']);
+        $this->assertEquals(true, $mailjetMessage['TemplateLanguage']);
+        $this->assertEquals('air-traffic-control@mailjet.com', $mailjetMessage['TemplateErrorReporting']);
+        $this->assertEquals('deliver', $mailjetMessage['TemplateErrorDeliver']);
+        $this->assertEquals(3, $mailjetMessage['Priority']);
+        $this->assertEquals('azertyuiop', $mailjetMessage['CustomCampaign']);
+        $this->assertEquals(false, $mailjetMessage['DeduplicateCampaign']);
+        $this->assertEquals(1, $mailjetMessage['TrackOpens']);
```
Comment (arnaudbreton) [inline Tests/SwiftMailer/MailjetTransportv31Test.php]: It's a string in Send API v3.1, see https://dev.mailjet.com/beta/#send-api-json-properties

File: Tests/SwiftMailer/MailjetTransportv31Test.php [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+        $this->assertEquals(2, $mailjetMessage['TrackClicks']);
```
Comment (arnaudbreton) [inline Tests/SwiftMailer/MailjetTransportv31Test.php]: It's a string in Send API v3.1, see https://dev.mailjet.com/beta/#send-api-json-properties

**Review by arnaudbreton — APPROVED** [review]

**Comment by latanasov** [comment]
@Nightbr  one more time thanks for your recommendations, they have been implemented with the recent [commit ](https://github.com/mailjet/MailjetSwiftMailer/commit/d3604c21ef6b0cfb5ec2152dfd3f1afdf8933bb4).

**Pull request merged**